tests/test_formatted_text.py

```diff
@@ -103,6 +103,66 @@ def test_ansi_true_color():
     ]
 
 
+def test_ansi_interpolation():
+    # %-style interpolation.
+    value = ANSI("\x1b[1m%s\x1b[0m") % "hello"
```

Comment:
For the tests, it would be good to have at least a ``\x1b`` in the right hand argument, so that we can escaping happens properly.